src/testing/storage.zig

```diff
@@ -303,8 +303,10 @@ pub const Storage = struct {
             storage.write_sectors_finish(write);
         }
 
-        // Process the queues in a single loop, since their callbacks may append to each other.
-        while (storage.next_tick_queue.pop()) |next_tick| {
+        // Avoid chained completions from extending how long it's popped.
+        var q = storage.next_tick_queue;
+        storage.next_tick_queue.reset();
+        while (q.pop()) |next_tick| {
             next_tick.callback(next_tick);
         }
```

Comment:
This commit fixed the seed `13070689082331695065`, which causes the simulator to loop indefinitely (not during the liveness check), because we finished the prefetch of invalid filters using `next_tick`.

I want it to be exhaustively fuzzed by the CFO because I'm not 100% confident that this fix reproduces exactly the same behavior of the real Storage `on_next_tick`, but apparently, the previous code didn't either. 🤣

Reply:
Reverted the code since it crashed [another test](https://github.com/tigerbeetle/tigerbeetle/actions/runs/9585722422/job/26432244243).
This seed doesn't fail anymore, so let it fuzz for some time!